contributors/devel/sig-architecture/feature-gates.md

```diff
@@ -0,0 +1,61 @@
+# Feature gate progression in Kubernetes
+
+Feature flags are how Kubernetes tracks progression of functionality that is available when deployed.
+
+## Lifecycle
+
+Feature progress through `Alpha` -> `Beta` -> `GA`. Sometimes we end up deciding that a feature is not going to be supported and we end up marking them as `Deprecated`.
+
+When we add a feature flag, we basically add if/else conditions to ensure that a feature is ONLY activated when either the default is on or if the deployer has switched it on explicitly.
```

Comment:
Something like "When a feature gate is disabled, the system should behave as if the feature doesn't exist.  The only exception to this  is API input validation on updates, which should preserve and validate data if and only if it was present before the update (which could occur in case of a version rollback)."

@liggitt Every time I write this out I question it.  We don't want this to turn into a design doc,. but explaining it might be useful?